DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/ipv4/test_ipv4_routing.py

```diff
@@ -0,0 +1,413 @@
+            "protocol": "ip",
+            "rate": "1000",  # pps
+            "srcMac": "aa:bb:cc:dd:ee:01",
+            "dstMac": mac_port0,
+            "srcIp": "1.1.1.2",
+            "dstIp": "16.0.0.0",
+        },
+        f"{tg_ports[1]} -> {tg_ports[0]}": {
+            "type": "raw",
+            "ip_source": dev_groups[tg_ports[1]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "protocol": "ip",
+            "rate": "1000",  # pps
+            "srcMac": "aa:bb:cc:dd:ee:02",
+            "dstMac": mac_port1,
+            "srcIp": "2.2.2.2",
+            "dstIp": "48.0.0.0",
+        },
+    }
+    await tgen_utils_setup_streams(tgen_dev, None, streams)
+
+    await tgen_utils_start_traffic(tgen_dev)
+    await asyncio.sleep(traffic_duration)
+    await tgen_utils_stop_traffic(tgen_dev)
+
+    stats = await tgen_utils_get_traffic_stats(tgen_dev, "Flow Statistics")
+    for row in stats.Rows:
+        assert tgen_utils_get_loss(row) == 0.000
+
+    await tgen_utils_stop_protocols(tgen_dev)
+
+
+@pytest.mark.asyncio
+async def test_ipv4_route_btw_vlan_devices(testbed):
+    """
+    Test Name: test_ipv4_route_btw_vlan_devices
+    Test Suite: suite_functional_ipv4
+    Test Overview: Test IPv4 route between vlan devices
+    Test Procedure:
+    1. Create a bridge entity
+    2. Enslave ports to bridge and create VLAN-devices
+    3. Set link up on all participant ports and VLAN-devices and add bridges to VLAN of the VLAN-devices
+    4. Configure ip address on VLAN-devices and then add ports to VLANs
+    5. Verify offload flag appear in VLAN-devices default routes
+    6. Prepare streams from one VLAN-device`s neighbor to the other
+    7. Transmit Traffic
+    8. Verify traffic is forwarded to both VLAN-devices` neighbors
+    9. Remove IP address from VLAN-devices and re-configure; Send ARPs and transmit again
+    10. Verify offload flag appear in VLAN-devices default routes
+    11. Verify traffic is forwarded to both VLAN-devices` neighbors
+    """
+    tgen_dev, dent_devices = await tgen_utils_get_dent_devices_with_tgen(testbed, [], 4)
+    if not tgen_dev or not dent_devices:
+        print("The testbed does not have enough dent with tgen connections")
+        return
+    dent_dev = dent_devices[0]
+    dent = dent_dev.host_name
+    tg_ports = tgen_dev.links_dict[dent][0]
+    ports = tgen_dev.links_dict[dent][1]
+    traffic_duration = 10
+    bridge = "br0"
+    vlan10 = f"{bridge}.10"
+    vlan20 = f"{bridge}.20"
+
+    address_map = (
+        # swp port, vlan iface, tg port, swp ip,   tg ip,    plen, vid
+        (ports[0], vlan10, tg_ports[0], "1.1.1.1", "1.1.1.2", 24,  10),
+        (ports[1], vlan20, tg_ports[1], "2.2.2.1", "2.2.2.2", 24,  20),
+    )
+
+    # Enable IPv4 forwarding
+    rc, out = await dent_dev.run_cmd(f"sysctl -n net.ipv4.ip_forward=1")
+    assert rc == 0
+
+    # 1. Create a bridge entity
+    out = await IpLink.delete(input_data=[{dent: [
```

Comment:
I would expect that we have a clear device by default instead of removing config "just in case"